components/modal/modal.service.ts

```diff
@@ -134,13 +134,11 @@ export class NzModalService implements OnDestroy {
     const overlayConfig = new OverlayConfig({
       hasBackdrop: true,
       scrollStrategy: this.overlay.scrollStrategies.block(),
+      backdropClass: getValueWithConfig(config.nzMask, globalConfig.nzMask, true) ? MODAL_MASK_CLASS_NAME : '',
```

Comment:
```suggestion
      hasBackdrop: getValueWithConfig(config.nzMask, globalConfig.nzMask, true),
      scrollStrategy: this.overlay.scrollStrategies.block(),
```

Reply:
Sure, I've resubmitted the code, please take a look.

Reply:
There is a test that failed, can you take a look?

Reply:
In the test cases, there is a dynamic modification of the global configuration nzMask, but the CDK overlay does not have an API to dynamically add or remove the backdrop, so it cannot satisfy all test cases. Therefore, it was modified to 
``` typescript
{
  hasBackdrop: true,
  backdropClass: getValueWithConfig(config.nzMask, globalConfig.nzMask, true) ? MODAL_MASK_CLASS_NAME : ''
}
```